lib/puppet/provider/service/systemd.rb

```diff
@@ -40,6 +40,8 @@ def disable
     output = systemctl(:disable, @resource[:name])
   rescue Puppet::ExecutionFailure
     raise Puppet::Error, "Could not disable #{self.name}: #{output}", $!.backtrace
+  ensure
+    @cached_enabled = nil
```

Comment:
It might be simpler to pull this out into a helper for calls to systemctl. I wrote it, but it makes me a little nervous.